src/rgw/driver/rados/rgw_rados.h

```diff
@@ -423,6 +430,9 @@ class RGWRados
   using RGWChainedCacheImpl_bucket_info_entry = RGWChainedCacheImpl<bucket_info_entry>;
   RGWChainedCacheImpl_bucket_info_entry *binfo_cache;
 
+  using RGWChainedCacheImpl_bucket_topics_entry = RGWChainedCacheImpl<pubsub_bucket_topics_entry>;
+  RGWChainedCacheImpl_bucket_topics_entry *topic_cache;
```

Comment:
i think this is the cause for the crash. in radosgw-admin the RGWRados class is created using `init_raw_storage_provider()` which does not allocate the `topic_cache`.
this is ok, however, when the topic cache is deleted, since it is not initialized to `nullptr`, it will cause a crash.

> note that when deleting, it is unnecessary to check if the pointer was allocated or not, just make sure it is initialized to `nullptr` (it is safe to delete `nullptr`)